### DISCUSSION TITLE
iOS application RedrawRequested spam

### DISCUSSION THREAD

**Discussion by nikolajevs86** [bevyengine/bevy#8884] [Q&A]
Hi! I am testing iOS app using bevy and it generate a lot of spam:
WARN winit::platform_impl::platform::app_state: processing non `RedrawRequested` event after the main event loop: RedrawEventsCleared

These warns does not affect the rendering and app is shown as it expected, but it really hard to debug other app logs as it generates a tons of such messages.

Does somebody knows how I can prevent this notifications?

**Comment by NiklasEi** [comment]
This seems to be a problem in winit. rust-windowing/winit#2911 contains changes that you could add to your own winit fork to prevent these logs. Winit seems to plan to remove the `RedrawEventsCleared` event according to rust-windowing/winit#2900, which should also remove this log in the mid term.

If you don't want to mess around with custom forks or wait until this is fixed in winit, you might be able to filter out these logs specifically. There is [an example on how to apply log filters in Bevy](https://github.com/bevyengine/bevy/blob/v0.9.1/examples/app/logs.rs). Looking at [the tracing documentation about the filters,](https://docs.rs/tracing-subscriber/0.3.17/tracing_subscriber/filter/struct.EnvFilter.html) it should be possible to filter this message out using a filter target like `winit::platform_impl::platform::app_state`.
